robonomicsinterface/classes/subscriptions.py

```diff
@@ -81,12 +93,22 @@ def _event_callback(self, index_obj: tp.Any, update_nr: int, subscription_id: in
         chain_events: list = self._custom_functions.chainstate_query("System", "Events")
         for events in chain_events:
             if events["event_id"] in self._event.value:
-                if self._target_address is None:
-                    self._callback(events["event"]["attributes"])  # All events
-                elif (
+                should_callback = (self._target_address is None) or (
                     events["event"]["attributes"][
-                        0 if self._event in [SubEvent.NewRecord, SubEvent.TopicChanged, SubEvent.NewDevices] else 1
+                        0 if self._event in [
+                            SubEvent.NewRecord,
+                            SubEvent.TopicChanged,
+                            SubEvent.NewDevices,
+                        ]
+                        else 1
                     ]
                     in self._target_address
-                ):
-                    self._callback(events["event"]["attributes"])  # address-targeted
+                )
+                if not should_callback:
```

Comment:
What do you think on simply adding the `event_id` to the output data tuple as a tailing or heading element?
`output = events["event"]["attributes"] + (event_id,)`

Reply:
It will introduce a breaking change for destructing assignments. For instance `NewLaunch` callback may look like:

```python
def callback(data):
    sender, _, _ = data
```

Reply:
Could you explain?
The `NewLaunch` looks like `sender, target, command` right now, I offer to add `event_id` as a 4th element.